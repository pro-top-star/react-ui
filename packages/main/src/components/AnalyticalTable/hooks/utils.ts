export const makeTemplateColumns = (columns, resizedColumns) => {
  return columns
    .map((column) => {
      if (resizedColumns[column.id]) {
        return `${resizedColumns[column.id]}px`;
      }
      return `minmax(${column.minWidth}px, ${column.width || '1fr'})`;
    })
    .join(' ');
};
