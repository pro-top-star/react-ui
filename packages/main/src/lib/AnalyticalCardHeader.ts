import { AnalyticalCardHeader } from '../components/AnalyticalCard/header/AnalyticalCardHeader';

export { AnalyticalCardHeader };
