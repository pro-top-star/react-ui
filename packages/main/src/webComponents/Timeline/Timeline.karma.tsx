import { mountThemedComponent } from '@shared/tests/utils';
import { expect, use } from 'chai';
import { matchSnapshot } from 'chai-karma-snapshot';
import React from 'react';
import { Timeline } from '../../lib/Timeline';

use(matchSnapshot);

describe('Timeline', () => {
  it('Basic Test (generated)', () => {
    const wrapper = mountThemedComponent(<Timeline />);
    expect(wrapper.debug()).to.matchSnapshot();
  });
});
