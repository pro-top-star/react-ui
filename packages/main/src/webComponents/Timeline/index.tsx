import { withWebComponent } from '@ui5/webcomponents-react/lib/withWebComponent';
import UI5Timeline from '@ui5/webcomponents/dist/Timeline';
import React, { FC, ReactNode } from 'react';
import { WithWebComponentPropTypes } from '../../internal/withWebComponent';

export interface TimelinePropTypes extends WithWebComponentPropTypes {
  children?: ReactNode | ReactNode[];
}

const Timeline: FC<TimelinePropTypes> = withWebComponent<TimelinePropTypes>(UI5Timeline);

Timeline.displayName = 'Timeline';

export { Timeline };
