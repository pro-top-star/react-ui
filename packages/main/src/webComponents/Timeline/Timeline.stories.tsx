import '@ui5/webcomponents-icons/dist/icons/calendar';
import '@ui5/webcomponents-icons/dist/icons/phone';
import { Timeline } from '@ui5/webcomponents-react/lib/Timeline';
import { TimelineItem } from '@ui5/webcomponents-react/lib/TimelineItem';
import React from 'react';

export default {
  title: 'UI5 Web Components / Timeline',
  component: Timeline,
  parameters: {
    subcomponents: { TimelineItem }
  },
  args: {
    titleText1: 'called',
    subtitleText1: '2019/01/01',
    icon1: 'phone',
    itemName1: 'John Smith',
    titleText2: 'Weekly Sync - CP Design',
    subtitleText2: '2019/01/02',
    icon2: 'calendar'
  },
  argTypes: {
    ref: {
      type: null
    },
    children: {
      type: null
    }
  }
};

export const generatedDefaultStory = (props) => (
  <Timeline>
    <TimelineItem
      titleText={props.titleText1}
      subtitleText={props.subtitleText1}
      icon={props.icon1}
      itemName={props.itemName1}
    />
    <TimelineItem titleText={props.titleText2} subtitleText={props.subtitleText2} icon={props.icon2}>
      <div>MR SOF02 2.43</div>
    </TimelineItem>
  </Timeline>
);

generatedDefaultStory.storyName = 'Generated default story';
