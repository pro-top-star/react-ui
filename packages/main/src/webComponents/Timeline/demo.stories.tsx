import '@ui5/webcomponents-icons/dist/icons/calendar';
import '@ui5/webcomponents-icons/dist/icons/phone';
import { Timeline } from '@ui5/webcomponents-react/lib/Timeline';
import { TimelineItem } from '@ui5/webcomponents-react/lib/TimelineItem';
import React from 'react';

export default {
  title: 'UI5 Web Components / Timeline',
  component: Timeline,
  parameters: {
    subcomponents: { TimelineItem }
  }
};

export const generatedDefaultStory = () => (
  <Timeline>
    <TimelineItem titleText="called" subtitleText={'2019/01/01'} icon="phone" itemName="John Smith" />
    <TimelineItem titleText="Weekly Sync - CP Design" subtitleText={'2019/01/02'} icon="calendar">
      <div>MR SOF02 2.43</div>
    </TimelineItem>
  </Timeline>
);

generatedDefaultStory.story = {
  name: 'Generated default story'
};
