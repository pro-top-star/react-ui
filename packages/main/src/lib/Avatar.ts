import { Avatar } from '../webComponents/Avatar';

export { Avatar };
