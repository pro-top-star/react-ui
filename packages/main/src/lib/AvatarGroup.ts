import { AvatarGroup } from '../webComponents/AvatarGroup';
import type { AvatarGroupPropTypes } from '../webComponents/AvatarGroup';

export { AvatarGroup };
export type { AvatarGroupPropTypes };
