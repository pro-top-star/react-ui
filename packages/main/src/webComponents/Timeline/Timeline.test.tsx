import { mountThemedComponent } from '@shared/tests/utils';
import { Timeline } from '@ui5/webcomponents-react/lib/Timeline';
import React from 'react';

describe('Timeline', () => {
  test('Basic Test (generated)', () => {
    const wrapper = mountThemedComponent(<Timeline />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
