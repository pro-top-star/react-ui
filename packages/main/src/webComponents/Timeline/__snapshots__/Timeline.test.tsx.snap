// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline Basic Test (generated) 1`] = `<ui5-timeline />`;
